session.go

```diff
@@ -991,27 +995,57 @@ func (s *session) loadCommonGlobalVariablesIfNeeded() error {
 	return nil
 }
 
-type txnWithErr struct {
+// txnFuture is a promise, which promises to return a txn in future.
+type txnFuture struct {
```

Comment:
Since you defined the type, why not define a method wait and get the transaction?